src/sentry/deletions/defaults/group.py

```diff
@@ -204,19 +206,45 @@ def delete_events_from_nodestore(self, events: Sequence[Event]) -> None:
     def delete_events_from_snuba(self) -> None:
         requests = []
         for project_id, group_ids in self.project_groups.items():
-            query = DeleteQuery(
-                self.dataset.value,
-                column_conditions={"project_id": [project_id], "group_id": list(group_ids)},
-            )
-            request = Request(
-                dataset=self.dataset.value,
-                app_id=self.referrer,
-                query=query,
-                tenant_ids=self.tenant_ids,
-            )
-            requests.append(request)
+            # Split group_ids into batches where the sum of times_seen is less than max_rows_to_delete
+            current_batch: list[int] = []
+            current_batch_rows = 0
+
+            # Get times_seen for each group
+            groups = Group.objects.filter(id__in=group_ids).values("id", "times_seen")
+            group_times_seen = {g["id"]: g["times_seen"] for g in groups}
```

Comment:
Instead of refetching here, should we just store this when we have the group here? https://github.com/getsentry/sentry/blob/b04bd3e6b08e5a0f8542c1c1fdf2648bbabd80bb/src/sentry/deletions/defaults/group.py#L82-L84

Reply:
Good idea! I addressed your feedback.